pkg/injector/patch.go

```diff
@@ -66,165 +49,46 @@ func (wh *webhook) createPatch(pod *corev1.Pod, namespace string, proxyUUID uuid
 
 	enableMetrics, err := wh.isMetricsEnabled(namespace)
 	if err != nil {
 		log.Error().Err(err).Msgf("Error checking if namespace %s is enabled for metrics", namespace)
 		return nil, err
 	}
 	if enableMetrics {
-		// Patch annotations
-		prometheusAnnotations := map[string]string{
-			constants.PrometheusScrapeAnnotation: strconv.FormatBool(true),
-			constants.PrometheusPortAnnotation:   strconv.Itoa(constants.EnvoyPrometheusInboundListenerPort),
-			constants.PrometheusPathAnnotation:   constants.PrometheusScrapePath,
+		if pod.Annotations == nil {
+			pod.Annotations = make(map[string]string)
 		}
-		patches = append(patches, updateMapType(
-			pod.Annotations,
-			prometheusAnnotations,
-			annotationsBasePath)...,
-		)
+		pod.Annotations[constants.PrometheusScrapeAnnotation] = strconv.FormatBool(true)
+		pod.Annotations[constants.PrometheusPortAnnotation] = strconv.Itoa(constants.EnvoyPrometheusInboundListenerPort)
+		pod.Annotations[constants.PrometheusPathAnnotation] = constants.PrometheusScrapePath
 	}
 
 	// This will append a label to the pod, which points to the unique Envoy ID used in the
 	// xDS certificate for that Envoy. This label will help xDS match the actual pod to the Envoy that
 	// connects to xDS (with the certificate's CN matching this label).
-	labelsToAdd := map[string]string{constants.EnvoyUniqueIDLabelName: proxyUUID.String()}
-	patches = append(patches, updateMapType(
-		pod.Labels,
-		labelsToAdd,
-		labelsBasePath)...,
-	)
-
-	return json.Marshal(patches)
-}
-
-func addVolume(target, add []corev1.Volume, basePath string) (patch []JSONPatchOperation) {
-	isFirst := len(target) == 0 // target is empty, use this to create the first item
-	var value interface{}
-	for _, volume := range add {
-		value = volume
-		volumePath := basePath
-		if isFirst {
-			isFirst = false
-			value = []corev1.Volume{volume}
-		} else {
-			volumePath += "/-"
-		}
-		patch = append(patch, JSONPatchOperation{
-			Op:    addOperation,
-			Path:  volumePath,
-			Value: value,
-		})
-	}
-	return patch
-}
-
-func addContainer(target, add []corev1.Container, basePath string) (patch []JSONPatchOperation) {
-	isFirst := len(target) == 0 // target is empty, use this to create the first item
-	var value interface{}
-	for _, container := range add {
-		value = container
-		path := basePath
-		if isFirst {
-			isFirst = false
-			value = []corev1.Container{container}
-		} else {
-			path += "/-"
-		}
-		patch = append(patch, JSONPatchOperation{
-			Op:    addOperation,
-			Path:  path,
-			Value: value,
-		})
-	}
-	return patch
-}
-
-// createMapPatchOperation is used specifically for the very first 'add' operation
-// for a non-existent 'path'.
-func createMapPatchOperation(key, path, value string) JSONPatchOperation {
-	return JSONPatchOperation{
-		Op:   addOperation,
-		Path: path,
-		Value: map[string]string{
-			key: value,
-		},
-	}
-}
+	pod.Labels[constants.EnvoyUniqueIDLabelName] = proxyUUID.String()
```

Comment:
Can `pod.Labels` be nil? I see a nil check for annotations but not for labels.

Reply:
Thanks for catching this!